idx/memory/memory.go

```diff
@@ -435,6 +461,12 @@ func (m *MemoryIdx) Delete(orgId int, pattern string) ([]schema.MetricDefinition
 	if err != nil {
 		return nil, err
 	}
+
+	// by deleting one or more nodes in the tree, any defs that previously failed may now
+	// be able to be added. An easy way to support this is just reset this map and give them
+	// all a chance again
+	m.FailedDefs = make(map[string]error)
```

Comment:
I am not sure this is the best approach. Ideally in future we will expose the Delete API to users.  On a large multi tenant system we could potentially see a steady stream of deletes coming in.

But with that said, i dont have a better suggestion and this certainly solves the problem we have today.